libs/libc/machine/Kconfig

```diff
@@ -44,6 +44,10 @@ config ARCH_ROMGETC
 # Default settings for C library functions that may be replaced with
 # architecture-specific versions.
 
+config LIBC_ARCH_ATOMIC
```

Comment:
The implementation for the atomic builtins should not be optional.
Instead, the OS-supported implementation should be used by every architecture that does not provide instructions for atomic memory operations.
We should create a hidden `bool` config (e.g. `ARCH_HAVE_ATOMICS`) and those ISA that do not provide (armv6-m, RISC-V without the `A` extension...) would be the ones to **not** select it.